packages/core/src/types.ts

```diff
@@ -19,6 +19,7 @@ export const enum KeyDerivationMethod {
 export interface WalletConfig {
   id: string
   key: string
+  rekey?: string
```

Comment:
I think this shouldn't be part of the open method, but rathe as a separate rotate wallet key config

Reply:
Agree but we have a problem here, I could create a different interface or type which will move around as long as I am in AFJ but the "this.indy.openWallet( ..." in IndyWallet module take a "WalletConfig" type and I can't go lower than that its part of the indy-sdk types repo

Or do you mean we create the lets say "WalletConfigRekey" in indy-sdk types repo? I don't think this is good idea since we still only have one function to open the wallet "openWallet" in the LibIndy wrappers and it already takes an optional argument "rekey" as you mentioned before. lots of workI guess

Reply:
Agree but we have a problem here, I could create a different interface or type which will move around as long as I am in AFJ but the "this.indy.openWallet( ..." in IndyWallet module take a "WalletConfig" type and I can't go lower than that its part of the indy-sdk types repo

Or do you mean we create the lets say "WalletConfigRekey" in indy-sdk types repo? I don't think this is good idea since we still only have one function to open the wallet "openWallet" in the LibIndy wrappers and it already takes an optional argument "rekey" as you mentioned before. lots of workI guess

Reply:
Yeah agree changing indy-sdk is not a good idea, but I think we can hide it in AFJ.

The implementation of the rotateKey method could like something like this:

```ts
// could optionally call this.close() here, but this makes it more explicit on how to do key rotation
if (this.wallet.isInitialized) {
  throw new AriesFrameworkError("Wallet must be closed to perform key rotation. Call agent.shutdown() before rotating your wallet key")
}

// we create a private method ._open in the indy wallet that is called by .open() and .rotateKey()
// This way .open() in the public interface doesn't have the rekey property
this._open({

})
```

Reply:
Sure, 
But the 'rekey' has to be in the 'OpenWalletCredentials' interface so if you can review the Pr [there](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59256)